service/resource/cloudformation/create.go

```diff
@@ -1,7 +1,63 @@
 package cloudformation
 
-import "context"
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/microerror"
+)
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+	stackInput, ok := createChange.(awscloudformation.CreateStackInput)
+	if !ok {
+		return microerror.Mask(fmt.Errorf("unexpected type, expecting %T, got %T", stackInput, createChange))
+	}
+
+	_, err := r.awsClient.CreateStack(&stackInput)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return awscloudformation.CreateStackInput{}, microerror.Mask(err)
+	}
+
+	currentStackState, ok := currentState.(StackState)
+	if !ok {
+		return awscloudformation.CreateStackInput{}, microerror.Mask(fmt.Errorf("unexpected type, expecting %T, got %T", currentStackState, currentState))
+	}
+	desiredStackState, ok := desiredState.(StackState)
+	if !ok {
+		return awscloudformation.CreateStackInput{}, microerror.Mask(fmt.Errorf("unexpected type, expecting %T, got %T", desiredStackState, desiredState))
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out if the main stack should be created")
+
+	createState := awscloudformation.CreateStackInput{
+		StackName: aws.String(""),
+	}
+
+	if currentStackState.Name != desiredStackState.Name {
```

Comment:
This here is tricky IMO. It depends on the resource how to manage the create/update events. I would say for a create event we should only check if there is a name. Comparing the name against a specific value (the desired state here) should be done during the update event. That way we separate the semantics of creating and updating. Creation is binary, a resource is there or not. Updates might be exponentially more complicated. We have to check for all kinds of values and if they match our desired values.